File: oa-portal/problems/m26-metro-with-passes/starters/main.java
import java.io.*;
import java.util.*;

public class Main {

    /* links[i] = {u, v, cost}, two-way. Return -1 if station n is unreachable. */
    static long cheapestFare(int n, int k, long[][] links) {
        // write your code here

        return -1;
    }

    public static void main(String[] args) throws IOException {
        FastReader in = new FastReader();
        int n = in.nextInt(), m = in.nextInt(), k = in.nextInt();
        long[][] links = new long[m][3];
        for (int i = 0; i < m; i++) {
            links[i][0] = in.nextLong(); links[i][1] = in.nextLong(); links[i][2] = in.nextLong();
        }

        System.out.println(cheapestFare(n, k, links));
    }

    static class FastReader {
        private final DataInputStream in = new DataInputStream(new BufferedInputStream(System.in, 1 << 16));
        int nextInt() throws IOException { return (int) nextLong(); }
        long nextLong() throws IOException {
            int b = in.read();
            while (b != '-' && (b < '0' || b > '9')) b = in.read();
            boolean neg = b == '-';
            if (neg) b = in.read();
            long v = 0;
            while (b >= '0' && b <= '9') { v = v * 10 + (b - '0'); b = in.read(); }
            return neg ? -v : v;
        }
        String next() throws IOException {
            int b = in.read();
            while (b <= ' ') b = in.read();
            StringBuilder sb = new StringBuilder();
            while (b > ' ') { sb.append((char) b); b = in.read(); }
            return sb.toString();
        }
    }
}

File: oa-portal/problems/m26-metro-with-passes/starters/main.cpp
#include <bits/stdc++.h>
using namespace std;

/* links[i] = {u, v, cost}, two-way. Return -1 if station n is unreachable. */
long long cheapestFare(int n, int k, const vector<array<long long,3>>& links) {
    // write your code here

    return -1;
}

int main() {
    int n, m, k;
    if (scanf("%d %d %d", &n, &m, &k) != 3) return 0;
    vector<array<long long,3>> links(m);
    for (auto &e : links) scanf("%lld %lld %lld", &e[0], &e[1], &e[2]);

    printf("%lld\n", cheapestFare(n, k, links));
    return 0;
}
